### ISSUE TITLE
Unable to recover LSI 9206-16e 

### ISSUE THREAD

**Issue by UltraHKR** [marcan/lsirec#19]
Hi Héctor!

First thank you for doing awesome work with lsirec & Apple Mx Linux development.

I'm grasping at straws here, I got for a very good price a (Dell?, IBM?, Lenovo? IDK the original rebrand, I only know it was broken by the user who sold it) LSI 9206-16e as you know this has 2x 2308 chips.

Chip #0 works flawlessly I can update and properly use sas2flash (BIOS & UEFI), lsiutil, lsirec, etc...
(I'm able to use almost all lsirec functions) 

Chip #1 is recognized I can do megarec -writesbr, -clean flash, etc.. But sas2flash (DOS & UEFI) is unable to flash, hostboot, I tried some different 2308 firmwares but it doesn't work... (Also tried flashing sbrempty.bin & 512bsbr.bin) 

With sas2flash It quite simply never leaves RESET. 

Using lsirec, it never leaves RESET no matter what I do. 

This is the result of lsirec bus-id -bind and info. 

Is there someway to fix this? (You and foddesha are the very the few people that I now, with deep enough knowledge of LSI FW & HW that can possibly know what's going on) [If needed I can use a flash programmer to deal with the 8-pin chips, in a week or so...) 

![IMG_20230627_060434.jpg](https://github.com/marcan/lsirec/assets/49404533/12e65d91-77f0-4b9a-8ed1-dd469b08c69c)

**Comment by FriedCheese2006** [comment]
Realize it's a year later and dunno if you still have the card or got this figured out on your own, but I was stuck in the same place. I managed to get everything recovered:
https://forums.servethehome.com/index.php?threads/lsi-9206-16e-soft-brick.46171/post-449041

**Comment by UltraHKR** [comment]
Do you got at any point "Firmware xxx with BBBh code" at startup, before both controllers initialize?

Also the SBR file are the same between controller 1 & 2?

I still have the card I can only use 2 out of 4 ports.

Tomorrow I will check it out...

**Comment by FriedCheese2006** [comment]
Yeah, same SBR for both. I honestly ran into so many different errors going through this process that I couldn't list them out. At one point, I was in a similar state you mentioned in your original comment. I could see the controller with megarec and successfully erase/wrote from there, but sas2flash could talk to the controller and lsiutil kept bombing on the firmware flash. Writing the SBR linked in the thread and making sure I had the right copy of the Broadcom firmware was the magic sauce to get it recovered. I spent three days messing with the card and ended up with a dozen different copies of the firmware trying to get everything to work...it was maddening. I started over from scratch today and recovered both.

**Comment by UltraHKR** [comment]
https://www.reddit.com/r/homelab/s/1fjSDyGjbT

Here's the screenshot of the error...

I tried so hard to get the 2nd controller working I started looking at a soldering station and IC chip programmer...

I will check it out tomorrow... I have to teardown my TrueNAS host where most of my (NFS) VM's reside to get my hands on that card...

**Comment by FriedCheese2006** [comment]
Oh....that's you! I stumbled on that post at some point. I do not have the BIOS firmware installed. I'm using the card as pure passthrough from proxmox to TrueNAS so don't need it.
